java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -213,6 +214,8 @@ public String toString() {
   }
 
   public ProgramState addConstraint(SymbolicValue symbolicValue, Constraint constraint) {
+    Preconditions.checkState(!(symbolicValue instanceof RelationalSymbolicValue && constraint == BooleanConstraint.FALSE),
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1698)